### PR TITLE
DSR to SSR

### PR DESCRIPTION
**Pull request by adamgoth**
Updates the DSR to use the SSR contract. Renames things. Removes proxy contract logic from deposit/withdraw.

### CODE REVIEW

**Review by b-pmcg — APPROVED** [review]

**Comment by b-pmcg** [comment]
nice that we don't have to worry about setting up a proxy for SSR

File: packages/hooks/src/ssr/useSsrWithdraw.ts [hunk 1]
```diff
@@ -1,62 +1,36 @@
...
-  // Only enabled if user has a balance in the DSR and they have a proxy
-  const enabled = dsrData && dsrData?.userDsrBalance > 0n && !!dsProxy?.data?.dsProxyAddress;
-
-  // encode withdraw function call for dsProxy to execute
-  const encodedJoinData = setMax
-    ? encodeFunctionData({
-        functionName: 'exitAll',
-        abi: proxyActionsDsrABI,
-        args: [mcdJoinDaiAddress[chainId], mcdPotAddress[chainId]]
-      })
```
Comment (b-pmcg) [inline packages/hooks/src/ssr/useSsrWithdraw.ts:43]: I think the way to implement a similar function to `exitAll` would be something like call [maxWithdraw](https://github.com/hexonaut/savings-dai/blob/master/src/SavingsDai.sol#L317) to calculate the exact amount and pass that value to the withdraw function.

Comment (rafinskipg) [reply packages/hooks/src/ssr/useSsrWithdraw.ts:43]: yeah , +1 to what @b-pmcg  said here, we could keep the maxWithdraw

**Review by rafinskipg — APPROVED** [review]

File: packages/hooks/src/ssr/useSsrWithdraw.ts (hunk 1 above)
Comment (adamgoth) [reply packages/hooks/src/ssr/useSsrWithdraw.ts:43]: I actually had that in there initially, but it seemed to work just fine without it. It also requires more updates to the UI and token input to indicate max withdraw. We had the `setMax` option in the hooks, but the UI was never sending `setMax` as `true` to the hook.

I'll add a TODO and we add a separate story for it.

**Pull request merged**